### PR TITLE
Add support for ZRS disks in virtualdisk and manageddisk modules

### LINKED ISSUE #792: Unable to create ZRS managed disk
##### SUMMARY
It seems that current module azure_rm_manageddisk and azure_rm_virtualmachine do not allow the creation of ZRS Managed Disk, I got the following issue when trying with Premium_ZRS (same issue with StandardSSD_ZRS) :

2022-03-10 08:54:00,657 p=2096608 u=root n=ansible | fatal: [localhost]: FAILED! = {'attempts': 5, 'changed': false, 'msg': 'Error creating the managed disk: Azure Error: InvalidParameter\\nMessage: SKU Premium_ZRS is not available.\\nTarget: sku'}

Would it be possible to add this feature in these 2 modules ?

##### ISSUE TYPE
- Feature Idea

##### COMPONENT NAME
azure_rm_manageddisk
azure_rm_virtualmachine

### PR DESCRIPTION
**Pull request by andreadecorte**
##### SUMMARY
Fixes #792 

##### ISSUE TYPE
- Feature Pull Request

##### COMPONENT NAME
- azure_rm_manageddisk
- azure_rm_virtualmachine

##### ADDITIONAL INFORMATION
Added 2 tests. Note that ZRS are currently only supported in selected regions, so this test right now will fail in US East. I tested it successfully in West Europe

### CODE REVIEW

**Comment by Fred-sun** [comment]
@andreadecorte Please update the test case as follows, pipeline deploys the test region to be EATUS and does not support "StandardSSD_ZRS" and" Premium_ZRS"! I only changed what I wanted to update!Thank you very much!
```e.g.
Position 1：
 - name: Create new manged disk with  I(account_type=StandardSSD_ZRS)
   azure_rm_manageddisk:
       resource_group: "{{ resource_group }}"
       name: "md{{ rpfx }}4"
       storage_account_type: "StandardSSD_ZRS"
       disk_size_gb: 2
       location: westus2
   register: output
Position 2：
 - name: Create new manged disk with  I(account_type=Premium_ZRS)
   azure_rm_manageddisk:
       resource_group: "{{ resource_group }}"
       name: "md{{ rpfx }}5"
       storage_account_type: "Premium_ZRS"
       disk_size_gb: 2
       location: westus2
   register: output
Position 3：
 - name: Delete all managed disk
   azure_rm_manageddisk:
       resource_group: "{{ resource_group }}"
       name: "md{{ rpfx }}{{ item }}"
       managed_by: ''
       state: absent
   with_items:
       - 1
       - 2
       - 3
       - 4
       - 5

**Comment by Fred-sun** [comment]
@andreadecorte Please update the test case as follows, pipeline deploys the test region to be EATUS and does not support "StandardSSD_ZRS" and" Premium_ZRS"! I only changed what I wanted to update!Thank you very much!
```e.g.
Position 1：
 - name: Create new manged disk with  I(account_type=StandardSSD_ZRS)
   azure_rm_manageddisk:
       resource_group: "{{ resource_group }}"
       name: "md{{ rpfx }}4"
       storage_account_type: "StandardSSD_ZRS"
       disk_size_gb: 2
   register: output
Position 2：
 - name: Create new manged disk with  I(account_type=Premium_ZRS)
   azure_rm_manageddisk:
       resource_group: "{{ resource_group }}"
       name: "md{{ rpfx }}5"
       storage_account_type: "Premium_ZRS"
       disk_size_gb: 2
   register: output
Position 3：
 - name: Delete all managed disk
   azure_rm_manageddisk:
       resource_group: "{{ resource_group }}"
       name: "md{{ rpfx }}{{ item }}"
       managed_by: ''
       state: absent
   with_items:
       - 1
       - 2
       - 3
       - 4
       - 5

**Comment by andreadecorte** [comment]
thanks for the review @Fred-sun , updated managed_disk test as per suggestion

**Comment by Fred-sun** [comment]
@andreadecorte Because some locations do not support the *_ZRS account type, please update the content of "azure_test_minimal_manageddisk.yml" as follows. Thank you very much!
```e.g.
- name: Create virtual network
  azure_rm_virtualnetwork:
    resource_group: "{{ resource_group }}"
    name: "{{ network_name }}-disk"
    address_prefixes: "{{ network }}"
    location: westus2

- name: Create subnet
  azure_rm_subnet:
    resource_group: "{{ resource_group }}"
    name: "{{ subnet_name }}"
    address_prefix: "{{ subnet }}"
    virtual_network: "{{ network_name }}-disk"

- name: Create network interface
  azure_rm_networkinterface:
    resource_group: "{{ resource_group }}"
    name: "{{ interface_name }}-disk"
    virtual_network: "{{ network_name }}-disk"
    subnet: "{{ subnet_name }}"
    location: westus2

- name: Create minimal VM with defaults and a custom managed disk type
  azure_rm_virtualmachine:
    resource_group: "{{ resource_group }}"
    name: "{{ vm_name }}"
    admin_username: "testuser"
    ssh_password_enabled: false
    managed_disk_type: StandardSSD_ZRS
    public_ip_allocation_method: Disabled
    location: westus2
    network_interface_names:
      - name: "{{ interface_name }}-disk"
        resource_group: "{{ resource_group }}"
    ssh_public_keys:
      - path: /home/testuser/.ssh/authorized_keys
        key_data: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDfoYlIV4lTPZTv7hXaVwQQuqBgGs4yeNRX0SPo2+HQt9u4X7IGwrtXc0nEUm6LfaCikMH58bOL8f20NTGz285kxdFHZRcBXtqmnMz2rXwhK9gwq5h1khc+GzHtdcJXsGA4y0xuaNcidcg04jxAlN/06fwb/VYwwWTVbypNC0gpGEpWckCNm8vlDlA55sU5et0SZ+J0RKVvEaweUOeNbFZqckGPA384imfeYlADppK/7eAxqfBVadVvZG8IJk4yvATgaIENIFj2cXxqu2mQ/Bp5Wr45uApvJsFXmi+v/nkiOEV1QpLOnEwAZo6EfFS4CCQtsymxJCl1PxdJ5LD4ZOtP xiuxi.sun@qq.com"
    vm_size: Standard_B1ms
    virtual_network: "{{ network_name }}-disk"
    image:
      offer: UbuntuServer
      publisher: Canonical
      sku: 16.04-LTS
      version: latest
  register: vm_output

 - name: Assert status succeeded
   assert:
     that:
       - vm_output.changed

- name: Delete VM
  azure_rm_virtualmachine:
    resource_group: "{{ resource_group }}"
    name: "{{ vm_name }}"
    remove_on_absent: all_autocreated
    state: absent

**Comment by andreadecorte** [comment]
thanks @Fred-sun , updated the test as per suggestion

File: tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_minimal_manageddisk.yml [hunk 1]
```diff
@@ -0,0 +1,89 @@
+- name: Create virtual network
+  azure_rm_virtualnetwork:
+    resource_group: "{{ resource_group }}"
+    name: "{{ network_name }}-disk"
+    address_prefixes: "{{ network }}"
+    location: westus2
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_minimal_manageddisk.yml]:
```suggestion
    location: westeurope
```

File: tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_minimal_manageddisk.yml [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+    address_prefix: "{{ subnet }}"
+    virtual_network: "{{ network_name }}-disk"
+
+- name: Create network interface
+  azure_rm_networkinterface:
+    resource_group: "{{ resource_group }}"
+    name: "{{ interface_name }}-disk"
+    virtual_network: "{{ network_name }}-disk"
+    subnet: "{{ subnet_name }}"
+    location: westus2
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_minimal_manageddisk.yml]:
```suggestion
    location: westeurope
```

File: tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_minimal_manageddisk.yml [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+
+- name: Create minimal VM with defaults and a custom managed disk type
+  azure_rm_virtualmachine:
+    resource_group: "{{ resource_group }}"
+    name: "{{ vm_name }}"
+    admin_username: "testuser"
+    ssh_password_enabled: false
+    managed_disk_type: StandardSSD_ZRS
+    public_ip_allocation_method: Disabled
+    location: westus2
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_minimal_manageddisk.yml]:
```suggestion
    location: westeurope
```

File: tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_minimal_manageddisk.yml [hunk 4]
```diff
@@ -0,0 +1,89 @@
...
+    network_interface_names:
+      - name: "{{ interface_name }}-disk"
+        resource_group: "{{ resource_group }}"
+    ssh_public_keys:
+      - path: /home/testuser/.ssh/authorized_keys
+        key_data: "ssh-rsa AAAAB3NzaC1yc2EAAAADAQABAAABAQDfoYlIV4lTPZTv7hXaVwQQuqBgGs4yeNRX0SPo2+HQt9u4X7IGwrtXc0nEUm6LfaCikMH58bOL8f20NTGz285kxdFHZRcBXtqmnMz2rXwhK9gwq5h1khc+GzHtdcJXsGA4y0xuaNcidcg04jxAlN/06fwb/VYwwWTVbypNC0gpGEpWckCNm8vlDlA55sU5et0SZ+J0RKVvEaweUOeNbFZqckGPA384imfeYlADppK/7eAxqfBVadVvZG8IJk4yvATgaIENIFj2cXxqu2mQ/Bp5Wr45uApvJsFXmi+v/nkiOEV1QpLOnEwAZo6EfFS4CCQtsymxJCl1PxdJ5LD4ZOtP xiuxi.sun@qq.com"
+    vm_size: Standard_B1ms
```
Comment (Fred-sun) [inline tests/integration/targets/azure_rm_virtualmachine/tasks/azure_test_minimal_manageddisk.yml]:
```suggestion
    vm_size: Standard_D4s_v3
```

**Review by Fred-sun — COMMENTED** [review]
We need to change location and vm_size for service reasons

**Comment by xuzhang3** [comment]
@andreadecorte LGTM, thanks

**Pull request merged**